core/flyout_base.js

```diff
@@ -426,77 +426,60 @@ Blockly.Flyout.prototype.hide = function() {
 
 /**
  * Show and populate the flyout.
- * @param {!Array|!NodeList|string} xmlList List of blocks to show.
+ * @param {!Array|!NodeList|string} contentInfo List of contents to display in
+ *     the flyout or a string with the name of the dynamic category.
  *     Variables and procedures have a custom set of blocks.
  */
-Blockly.Flyout.prototype.show = function(xmlList) {
+Blockly.Flyout.prototype.show = function(contentInfo) {
   this.workspace_.setResizesEnabled(false);
   this.hide();
   this.clearOldBlocks_();
 
-  // Handle dynamic categories, represented by a name instead of a list of XML.
+  // Handle dynamic categories, represented by a name instead of a list.
   // Look up the correct category generation function and call that to get a
   // valid XML list.
-  if (typeof xmlList == 'string') {
+  if (typeof contentInfo == 'string') {
     var fnToApply = this.workspace_.targetWorkspace.getToolboxCategoryCallback(
-        xmlList);
+        contentInfo);
     if (typeof fnToApply != 'function') {
       throw TypeError('Couldn\'t find a callback function when opening' +
           ' a toolbox category.');
     }
-    xmlList = fnToApply(this.workspace_.targetWorkspace);
-    if (!Array.isArray(xmlList)) {
+    contentInfo = fnToApply(this.workspace_.targetWorkspace);
+    if (!Array.isArray(contentInfo)) {
       throw TypeError('Result of toolbox category callback must be an array.');
     }
   }
+  // contentInfo is either an array with JSON, an Array with xml, or a NodeList.
+  var parsedContent = Blockly.utils.toolbox.parseToolbox(contentInfo);
 
   this.setVisible(true);
   // Create the blocks to be shown in this flyout.
   var contents = [];
   var gaps = [];
   this.permanentlyDisabled_.length = 0;
-  var default_gap = this.horizontalLayout_ ? this.GAP_X : this.GAP_Y;
-  for (var i = 0, xml; (xml = xmlList[i]); i++) {
-    if (!xml.tagName) {
+  var defaultGap = this.horizontalLayout_ ? this.GAP_X : this.GAP_Y;
+  for (var i = 0, contentItem; (contentItem = parsedContent[i]); i++) {
+    if (!contentItem.contenttype) {
       continue;
     }
-    switch (xml.tagName.toUpperCase()) {
+    switch (contentItem.contenttype.toUpperCase()) {
       case 'BLOCK':
-        var curBlock = Blockly.Xml.domToBlock(xml, this.workspace_);
-        if (!curBlock.isEnabled()) {
-          // Record blocks that were initially disabled.
-          // Do not enable these blocks as a result of capacity filtering.
-          this.permanentlyDisabled_.push(curBlock);
-        }
-        contents.push({type: 'block', block: curBlock});
-        // This is a deprecated method for adding gap to a block.
-        // <block type="math_arithmetic" gap="8"></block>
-        var gap = parseInt(xml.getAttribute('gap'), 10);
-        gaps.push(isNaN(gap) ? default_gap : gap);
+        var blockInfo = /** @type {Blockly.utils.toolbox.BlockInfo} */ (contentItem);
```

Comment:
Let's keep the naming consistent, blockItem from contentItem, or a blockInfo from a contentInfo

Reply:
done